client/my-sites/marketplace/pages/marketplace-domain-upsell/index.tsx

```diff
@@ -95,6 +98,9 @@ function CalypsoWrappedMarketplaceDomainUpsell(): JSX.Element {
 
 	const onDomainSelect = async ( suggestion: DomainSuggestions.DomainSuggestion ) => {
 		const { product_slug, domain_name } = suggestion;
+		if ( ! product_slug ) {
+			throw new Error( 'Cannot select domain without product slug' );
```

Comment:
I'm guessing we're adding the runtime type check because we're not 100% TypeScript and can't prevent people from calling this function without `product_slug`?

Reply:
I have no idea, actually, but the types here suggest that sometimes `DomainSuggestion` will not have a product_slug. I suspect the types could be improved but as these types are fairly complex and used in many places, I didn't want to risk it.